src/data-connect/data-connect-api-client-internal.ts

```diff
@@ -198,6 +198,203 @@ export class DataConnectApiClient {
     const message = error.message || `Unknown server error: ${response.text}`;
     return new FirebaseDataConnectError(code, message);
   }
+
+  /**
+   * Converts JSON data into a GraphQL literal string.
+   * Handles nested objects, arrays, strings, numbers, and booleans.
+   * Ensures strings are properly escaped.
+   */
+  private objectToString(data: any): string {
+    if (typeof data === 'string') {
+      const escapedString = data
+        .replace(/\\/g, '\\\\') // Replace \ with \\
+        .replace(/"/g, '\\"');  // Replace " with \"
+      return `"${escapedString}"`;
+    }
+    if (typeof data === 'number' || typeof data === 'boolean' || data === null) {
+      return String(data);
+    }
+    if (validator.isArray(data)) {
+      const elements = data.map(item => this.objectToString(item)).join(', ');
+      return `[${elements}]`;
+    }
+    if (typeof data === 'object' && data !== null) {
+      // Filter out properties where the value is undefined BEFORE mapping
+      const kvPairs = Object.entries(data)
+        .filter(([, val]) => val !== undefined)
+        .map(([key, val]) => {
+          // GraphQL object keys are typically unquoted.
+          return `${key}: ${this.objectToString(val)}`;
+        });
+  
+      if (kvPairs.length === 0) {
+        return '{}'; // Represent an object with no defined properties as {}
```

Comment:
Hmm, I'm not sure there's a case for when this would happen

Reply:
I think this hits if you pass in an object with all `undefined` values
```
director: {
    name: undefined,
    age: undefined
}
```